src/server/auth/server/api_server.go

```diff
@@ -672,6 +683,13 @@ func (a *apiServer) CreateRoleBindingInTransaction(txnCtx *txnenv.TransactionCon
 	// Call Create, this will raise an error if the role binding already exists.
 	key := resourceKey(resource)
 	roleBindings := a.roleBindings.ReadWrite(txnCtx.SqlTx)
+	// TODO(acohen4): what should the collision semantics be here?
```

Comment:
Is it necessary to add a read here? If the role binding key has a uniqueness constraint, whatever transaction commits second should fail and the first transaction will succeed at creating the repo.

Reply:
in tests where we create multiple authenticated clients, each client creation triggers the pfsClient.ActivateAuth RPC. it seems that now whenever we the Creation fails, even though our code gracefully handles the error, the DB Tx is invalidated because part of it failed.

Reply:
Hmmm. If we just wrote SQL we could make it idempotent by using an `on conflict` clause, does the sql collections library support that?

Reply:
@actgardner good point

@Tryneus what do you think is the best solution here? Should we handle this which custom SQL, or add a collections method for Create and do nothing on conflict?

I suppose another alternative would be to see if we could get Postgres to keep the transaction valid even when one INSERT fails - although I'm not sure if that's possible

Reply:
Ah, yeah, we should change `collection.Create()` to not invalidate the transaction, if possible.  Fix in PR #6002.